chevah/brink
# [#4022] Update the build system to newer pip and setuptools.

Scope
=====

We need to have modern pip and setuptools in our build system so that if we find errors we can report them upstream and hope that someone will look at them,

right now we have ancient (4 to 5 years old) version.

with the newest python-package we get pip by default... so we should no longer need to install pip and setuptools

We also need to prepare the migration to python 3

Changes
=======

the deps was broken into multiple parts so that we don't have to install everything on each slave

add support for running python_3 tests ... for now the brink code is not yet ported to python3 ... but we should do that soon in a separate ticket.

the paver.sh script was simplified as it assumes that python already has pip and setuptoools

also, paver.sh was updated to add support for the test_py3

How to try and test the changes
===============================

reviewers: @alibotean @brunogola  @dumol
 
check that changes make sense
